lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -0,0 +1,648 @@
+}
+
+Bool PointConditionalDistribution::equals(const DistributionImplementation & other) const
+{
+  const PointConditionalDistribution* p_other = dynamic_cast<const PointConditionalDistribution*>(&other);
+  return p_other && (*this == *p_other);
+}
+
+/* String converter */
+String PointConditionalDistribution::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << PointConditionalDistribution::GetClassName()
+      << " name=" << getName()
+      << " dimension=" << getDimension()
+      << " distribution=" << distribution_
+      << " indices=" << conditioningIndices_
+      << " values=" << conditioningValues_;
+  return oss;
+}
+
+String PointConditionalDistribution::__str__(const String & offset) const
+{
+  OSS oss(false);
+  oss << offset << getClassName() << "(distribution = " << distribution_ << ", indices = " << conditioningIndices_ << ", values = " << conditioningValues_ << ")";
+  return oss;
+}
+
+/* Virtual constructor */
+PointConditionalDistribution * PointConditionalDistribution::clone() const
+{
+  return new PointConditionalDistribution(*this);
+}
+
+void PointConditionalDistribution::update()
+{
+  const UnsignedInteger fullDimension = distribution_.getDimension();
+  if (conditioningIndices_.getSize())
+    logNormalizationFactor_ = distribution_.getMarginal(conditioningIndices_).computeLogPDF(conditioningValues_);
+  if (!(logNormalizationFactor_ > std::log(getPDFEpsilon())))
+    throw InvalidArgumentException(HERE) << "Conditioning vector PDF value is too low";
+  nonConditioningIndices_ = conditioningIndices_.complement(fullDimension);
+  setDimension(nonConditioningIndices_.getSize());
+  marginalConditionedDistribution_ = distribution_.getMarginal(nonConditioningIndices_);
+  setDescription(marginalConditionedDistribution_.getDescription());
+
+  // enable simplified path
+  useSimplifiedVersion_ = hasSimplifiedVersion(simplifiedVersion_);
+
+  computeRange();
+  isAlreadyComputedMean_ = false;
+  isAlreadyComputedCovariance_ = false;
+}
+
+
+Point PointConditionalDistribution::decompose(CovarianceMatrix & C) const
+{
+  const Point mu(distribution_.getMean());
+  const CovarianceMatrix cov(distribution_.getCovariance());
+  const CovarianceMatrix cxx(distribution_.getMarginal(nonConditioningIndices_).getCovariance());
+  const CovarianceMatrix cyy(distribution_.getMarginal(conditioningIndices_).getCovariance());
+  const Point mux = mu.select(nonConditioningIndices_);
+  const Point muy = mu.select(conditioningIndices_);
+  Matrix cxy(conditioningIndices_.getSize(), nonConditioningIndices_.getSize());
+  for (UnsignedInteger i = 0; i < conditioningIndices_.getSize(); ++ i)
+    for (UnsignedInteger j = 0; j < nonConditioningIndices_.getSize(); ++ j)
+      cxy(i, j) = cov(conditioningIndices_[i], nonConditioningIndices_[j]);
+  const Matrix icyy(cyy.inverse()); // too bad we cannot get the cholesky factor from Normal
```

Comment:
In most cases distribution_.getMarginal(conditioningIndices_) will return a distribution which can be cast into an EllipticalDistribution and this way you can get the Cholesky factor of cyy. If the cast fails then you fall back to the inverse method.